### PR TITLE
Enabled module selection

### PR DESCRIPTION
**Pull request by LarsHoldijk**
This pull request contains a very minor change to the zoo module. 

In the current version of the zoo it is not possible to use the `module_name` parameter of model loader. Due to this it is only possible to have one model for each zoo enabled git repository. By enabling the user to specify the module name in the `get_model()` function this restriction is lifted.

With this change it is easier for developers to contribute models to the zoo.

### CODE REVIEW

**Comment by jonasrauber** [comment]
Sounds good to me, thanks! @bveliqi can you also have a look?
Unfortunately it looks like the tests broke due to some unrelated version conflict?!

**Comment by wielandbrendel** [comment]
@LarsHoldijk That's a great addition! I am not yet sure whether this is the best way to implement this, though. Implemented this way one would need several python files each described a certain model, right? Wouldn't it be easier to include an argument to `module.create()`? This way one has just one file - foolbox_model.py - and one loads the right model with certain keywords. We'd just need to include an exception handling for backwards compatibility (for create-functions without an argument).

@jonasrauber What do you think?

**Comment by jonasrauber** [comment]
I might have misunderstood what the PR does. We certainly want the files to always be called `foolbox_model.py`. I somehow assumed this code just makes sure that the module will be imported under a different name than the file name using this PR. Most likely that's not actually the case. Thanks for pointining this out @wielandbrendel, not sure I understand your alternative suggestion.

I think the use case that needs to be solved is importing several models (each from its own repository and thus with its own `foolbox_model.py`) under different names in one python environment.

**Comment by wielandbrendel** [comment]
@jonasrauber Take a look at https://github.com/bethgelab/defensive-distillation/blob/master/foolbox_model.py : I want that the `create` function takes an argument `variant` that determines which model variant (e.g. _vanilla_ or _defended_) shall be instantiated.

**Comment by jonasrauber** [comment]
@wielandbrendel yes, I get that and I am fine with that. It just doesn't solve the more general problem of using different defenses in the same python session, because only one can be called `foolbox_model`; however, I don't know how to fix this right now and ideas are welcome, @LarsHoldijk.

**Comment by bveliqi** [comment]
@jonasrauber looks good to me!

**Comment by wielandbrendel** [comment]
@jonasrauber You are talking about importing more than one model whereas this PR talks about defining more than one model in the same git repository. That's pretty orthogonal.

**Comment by jonasrauber** [comment]
True, I guess I also misunderstood the intention of the PR. Sorry :D 
@wielandbrendel I agree with you, adding one (arbitrary) optional arguments to the create function and passing it (them) to `zoo.get_model(url)` and from there to `model.create(…)` sounds like the better solution

**Comment by LarsHoldijk** [comment]
@wielandbrendel @jonasrauber I had originally set out to pass a variable number of parameters to `zoo.get_model()` using `**kwargs` and let this flow into `model.create()`. When instantiating the model the create function can use these parameters (in the form of a dictionary) to either select which model to instantiate or do more specialized stuff. 

The implementation in the PR was however already available in the code and was therefore easier to do without changing the zoo to much. I can give my old intentions a try if this would be preferred.

**Comment by wielandbrendel** [comment]
@LarsHoldijk I'd definitely vote for the keyword arguments! That's the most flexible implementation and is backwards compatibility with existing zoo models. It would be great if you could give this a shot!

**Comment by LarsHoldijk** [comment]
@wielandbrendel Perfect, I will give that a shot again then.

**Comment by wielandbrendel** [comment]
@LarsHoldijk Great, thanks a lot!

**Comment by jonasrauber** [comment]
👍

**Comment by LarsHoldijk** [comment]
@jonasrauber @wielandbrendel The latest changes should allow a developer of a zoo repository to define an arbitrary number of parameters for its `create()` function. Whenever someone else then wishes to evaluate this model he/she can use the same set of parameters when calling `get_model()`. I've made the changes in such a way that it should be backwards compatible with both earlier usage of the `get_model()` method and implementations of `create()`.

I've tested the implementation of the `create()` method in a private repository we will be using to open source our latest LVQ models. Being able to simply use parameters for the function very intuitive. If you wish to see how it can be used, I could quickly add you to the repository as well.

File: foolbox/zoo/model_loader.py [hunk 1]
```diff
@@ -39,7 +40,10 @@ def _import_module(path, module_name='foolbox_model'):
...
 class DefaultLoader(ModelLoader):
 
-    def load(self, path, module_name='foolbox_model'):
-        module = ModelLoader._import_module(path, module_name)
-        model = module.create()
+    def load(self, path, **kwargs):
+        module = ModelLoader._import_module(path)
+        if kwargs:  # Empty dictionaries evaluate fo false
+            model = module.create(**kwargs)
+        else:  # For backwards compatibility
```
Comment (jonasrauber) [inline foolbox/zoo/model_loader.py]: Are you sure this is needed?
Isn’t unpacking an empty dictionary equivalent anyway?

Comment (LarsHoldijk) [reply foolbox/zoo/model_loader.py]: You're absolutely right, I misinterpreted how python would handle the unpacking of the dictionary.

**Comment by LarsHoldijk** [comment]
@jonasrauber  @wielandbrendel Can I get an update on the status of this pull request? The current implementation should address all your previous concerns.

**Comment by wielandbrendel** [comment]
@LarsHoldijk Looks all good, now we just have to fix the CI checks. Parts of these are strange (some binary incompatibilities, @jonasrauber can you take a look or shall we ignore them right now?) and the two other problems are just formatting problems (flake8 at the bottom of the logs).

**Comment by jonasrauber** [comment]
I think right now, the test tools don't run correctly. This is not the fault of this PR and probably caused by some version change of pytest-cov or so. Nevertheless, I'd prefer to first fix this issue so that the tests run fine again on `master` before merging this, simply to avoid introducing new problems (e.g. unchecked lines) that we miss now because the checks are not actually executed.
If someone wants to look into it (maybe it's sufficient to change the version of some test package), I'd appreciate it. I won't get to it this week.
@LarsHoldijk Thanks for the PR, looking forward to merging it!

**Comment by jonasrauber** [comment]
I am about to fix the test issues so that we will finally be able to merge this PR. Thanks for your patience. There is one more thing: it seems you have removed the ability to pass an alternative filename. This possibility currently exists and I don't see a reason to remove it. Do you or was it an accident?

**Comment by LarsHoldijk** [comment]
@jonasrauber Thank you for looking into the failing tests and merging the pull request. Looking at it I must indeed have accidentally removed the module section from the load method, this must have happened when I switched to kwargs.

**Comment by jonasrauber** [comment]
okay, I fixed the tests in master and updated this PR to also support `module_name`

**Pull request merged**